### ISSUE TITLE
Possible bug when parallel=True

### ISSUE THREAD

**Issue by Slasher111** [numba/numba#6925]
```python
import numpy as np
from numba import njit

data_global = np.array([(1, 2), (3, 4)], dtype=[('Option1', 'i4'), ('Option2', 'i4')])

@njit(parallel=True)
def modify_structured_array_field(data):
    fields_gl = ('Option1', 'Option2')
    data[1][fields_gl[1]] = 6
    return(data)

new_data = modify_structured_array_field(data_global)

print(new_data)
```
This code fails when I use `@njit(parallel=True)` with an error `No definition for lowering <built-in function setitem>`.
But it works fine when I use `@njit` without parallelization. Is this a bug or I'm just doing something wrong?

**Comment by rawwar** [comment]
Hi @Slasher111 , this bug is fixed by #6801 .  you can try development packae by using `conda install -c numba/label/dev numba` to test it.

**Comment by stuartarchibald** [comment]
Thanks for the report @Slasher111 and thanks for debugging this @rawwar. I can't reproduce on mainline, looks like #6801 fixed it.

Closing this issue as it seems to be resolved. If this is not the case please re-open with a comment about any item that appears to be unresolved. Many thanks.
